src/Tutor.Core/Domain/KnowledgeMastery/SessionTracker.cs

```diff
@@ -14,28 +14,59 @@ public class SessionTracker : EventSourcedEntity
     public TimeSpan DurationOfFinishedSessions { get; private set; } = new(0);
     public bool HasUnfinishedSession => StartOfUnfinishedSession.HasValue;
 
-    public TimeSpan? DurationOfUnfinishedSession => HasUnfinishedSession ? LastActivity.Value - StartOfUnfinishedSession.Value : null;
+    public TimeSpan? DurationOfUnfinishedSession => HasUnfinishedSession ? DateTime.UtcNow - StartOfUnfinishedSession.Value : null;
     public DateTime? StartOfUnfinishedSession { get; private set; }
-    public DateTime? LastActivity { get; private set; }
+
+    public bool IsPaused { get; private set; }
+    public TimeSpan DurationOfAllPauses => DurationOfFinishedPauses + DurationOfUnfinishedPauses.GetValueOrDefault();
+    public TimeSpan DurationOfFinishedPauses { get; private set; } = new(0);
+    public TimeSpan? DurationOfUnfinishedPauses => IsPaused ? DateTime.UtcNow - LastPause.Value : null;
+    public DateTime? LastPause { get; private set; }
```

Comment:
This name is not accurate. First, it does not model the concept of a "Pause". Next, it does not maintain the last pause time as it is set to null when the pause is over.

Maybe `UnfinishedPauseStart`?